skbio/metadata/_interval.py

```diff
@@ -512,6 +520,30 @@ def __init__(self, upper_bound):
         # Indicates if the IntervalTree needs to be rebuilt.
         self._is_stale_tree = False
 
+        if copy_from is not None:
+            for interval in copy_from._intervals:
+                # Only need to shallow-copy `bounds` and `fuzzy`
+                # because their elements are immutable.
+                bounds_cp = interval.bounds[:]
+                fuzzy_cp = interval.fuzzy[:]
+                metadata_cp = copy.copy(interval.metadata)
+
+                self.add(bounds_cp, fuzzy=fuzzy_cp, metadata=metadata_cp)
+
+    def _rebuild_tree(method):
```

Comment:
The more strongly I feel like putting it outside the class, the more I look at it, because it is more logic. "it was working fine" should not stop us improving it?